### DISCUSSION TITLE
idea: automatically remove channels that are out of range

### DISCUSSION THREAD

**Discussion by DC7IA** [hmatuschek/qdmr#171] [Ideas]
An idea for an additional feature:

An option to automatically remove channels that are out of range when writing to a radio.

This allows maintaining a single codeplug for a variety of radios with different ranges.

For example a codeplug containing frequencies in the 2 m and 70 cm amateur radio bands can then be written to a 70 cm radio. The 2 m channels simply aren't used.

Like this:

```
The following changels are out of range. Do you want to omit them when writing to the radio?

[ Yes, all of them ] [ Select channels to omit ] [ Abort writing ]
```

Or maybe something similar.

**Comment by DC7IA** [comment]
I guess that's not too difficult to do given that there are already checks for channels that are out of range.

**Comment by hmatuschek** [comment]
This is more complicated that you think, as channels might be referenced in scan lists and DMR APRS configurations as revert channels. So I have to be very careful not to produce a inconsistent codeplug. It would be easier to simply set them to "RX only". The MD390 manufacturer CPS, however, will then stop working properly.

**Comment by DC7IA** [comment]
> The MD390 manufacturer CPS, however, will then stop working properly.

Then ask the user and then do the appropriate action?

Removing channels that are out of range before writing is a better solution long-term, I think. But I also think there are a lot more things that are more relevant for now. :D

**Comment by hmatuschek** [comment]
I've just enabled "Discussions" for this repository. Hopefully I can bundle also email discussions there. How about moving this issue there?
